lib/cc/engine/analyzers/file_thread_pool.rb

```diff
@@ -54,6 +54,13 @@ def thread_count
             DEFAULT_CONCURRENCY
           end
         end
+
+        def with_thread_abort_on_exceptions
+          t = Thread.new do
+            yield
+          end
+          (t.abort_on_exception = true) && t
```

Comment:
Our style guide allows the use of `and` to avoid these parenthesis. That said, I'd prefer this formatting:

```rb
def with_thread_abort_on_exceptions(&block)
  thread = Thread.new(&block)
  thread.abort_on_exception = true
  thread
end
```

FWIW